kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/engine/DefaultJsonHandlerSuite.scala

```diff
@@ -373,6 +373,25 @@ class DefaultJsonHandlerSuite extends AnyFunSuite with TestUtils with DefaultVec
     checkAnswer(actResult, expResult)
   }
 
+  test("read json files, some of which are empty") {
+    val testFiles = fsClient.listFrom(getTestResourceFilePath("json-files-with-empty/1.json"))
+    val actResult = jsonHandler.readJsonFiles(
+      testFiles,
+      new StructType()
+        .add("c1", StringType.STRING)
+        .add("c3", StringType.STRING),
```

Comment:
it may be nice to see here something more like:

- StringType.STRING
- LongType.LONG)
- BooleanType.BOOLEAN

(this is also related to the other comment, comes down to json file structure and contents)